File: PYTHON/Mhs.py
from Civitas import Civitas

class Mhs (Civitas):
    # kelas untuk menampung data mhs dan method belajar

    # atribut
    __nim = ""
    __semester = ""

    # konstruktor
    def __init__(self):
        self.__nim = ""
        self.__semester = ""

    # setter getter
    def setNim(self, nim):
        self.__nim = nim
    def setSemester(self, semester):
        self.__semester = semester
    
    def getNim(self):
        return self.__nim
    def getSemester(self):
        return self.__semester
    
    # method
    def belajar(self):
        print("gamau belajar mau nya maen")

File: CPP/Mhs.cpp
#include <iostream>
#include <string>
#include "Civitas.cpp"

using namespace std;

class Mhs : public Civitas {
    // attribute
    private:
        string nim, semester;
    
    // constructor
    public:
        Mhs() {
            this->nim = "";
            this->semester = "";
        }

        Mhs(string nim, string semester, string prodi) {
            this->nim = nim;
            this->semester = semester;
        }

        // getter and setter
        string getNim() {
            return this->nim;
        }

        string getSemester() {
            return this->semester;
        }

        void setNim(string nim) {
            this->nim = nim;
        }

        void setSemester(string semester) {
            this->semester = semester;
        }

        // method
        void belajar() {
            cout << "gamau belajar mau nya maen" << endl;
        }

        void printMhs() {
            printCivitas();
            cout << "NIM : " << this->nim << endl;
            cout << "Semester : " << this->semester << endl;
        }

        // destruktor
        ~Mhs() {}
};
